### DISCUSSION TITLE
When standardSCParity not n/2

### DISCUSSION THREAD

**Discussion by izznogood** [minio/minio#11425] [Q&A] [answered]
Hello,
I configured 3 servers with 4 disks each.
During my previous POCs, the configuration was as follows:

> "onlineDisks": 12,
            "rrSCData": 10,
            "rrSCParity: 2,
            "standardSCData: 6,
            "standardSCParity: 6

today with version RELEASE.2021-02-01T22-56-52Z
by default I have this:

> "onlineDisks": 12,
            "rrSCParity: 2,
            "standardSCParity: 4

I think the parity at 4 is perfect, it means we can lose 4 disks (one server) and still use the service.
(but I can't find a perfectly satisfactory answer)
cf: 
- https://docs.min.io/docs/minio-erasure-code-quickstart-guide.html
- https://github.com/minio/minio/tree/master/docs/erasure/storage-class
- https://github.com/minio/minio/blob/master/docs/distributed/DESIGN.md
- https://github.com/minio/minio/issues/10948#issuecomment-731509531

In your opinion, is this correct or is it better to force the standardSCParity to 6?
(yes the best would be 4 servers.... but a have only 3)
Thank you!

**Comment by ravindk89** [comment, marked as answer]
Hi izznogood! You've discovered a *very* recent change to the product :)

The [RELEASE.2021-01-30T00-20-58Z](https://github.com/minio/minio/releases/tag/RELEASE.2021-01-30T00-20-58Z) now defaults to  EC:4 for deployments with 8 or more drives. We selected this new default based on our continued growth of experience with balancing redundancy requirements against maximizing the possible usable storage. 

I'm in the process of updating documentation to discuss both the now-legacy and new behavior, because not all users will have upgraded right away. Furthermore, clusters w/ data retain their existing parity settings after upgrading - this only applies to new clusters. 

At the end of the day, parity selection is up to the user. It is a choice between balancing your redundancy requirements against the total usable storage you want. Since EC:6 = half your drives go to parity (0.5 ratio of usable storage), you will naturally have less total storage. With EC:4 on a 12 drive setup, you have a 0.67 ratio of usable storage. 

As one small correction - we have two quorums to consider. Read Quorum is the number of drives required to maintain read operations. Write Quorum is the number of drives required to maintain write operations. The formula works like this:

- Read Quorum: `ERASURE_SET_DRIVES-EC:N`
- Write Quorum: `(ERASURE_SET_DRIVES-EC:N)+1`

Since the number of drives required by Write Quorum satisfies Read Quorum, Write Quorum implies Read Quorum.

Without getting too in the weeds on erasure set sizing, a 12-drive deployment will have a single 12-drive erasure set.

At EC:6, you need 7 drives for R+W, or 6 drives for R

At EC:4, you need 9 drives for R+W, or 8 drives for R

So in your scenario, the loss of a single server would result in 4 down drives - so you would lose write quorum at EC:4, but retain read quorum.

I hope this helps - I am working on substantial changes to our documentation to better capture some of the complexity around erasure coding and parity sizing.

**Reply by izznogood** [reply]
it's perfect, very clear, thank's a lot!!!!
I have all the information I need to continue my deployment :)
